### PR TITLE
Propose that MetalLB apply to the CNCF Sandbox

### PR DESCRIPTION
**Pull request by russellb**

### CODE REVIEW

**Comment by russellb** [comment]
@danderson @daxmc99 @johananl @rata

File: design/cncf-sandbox-proposal.md
Comment (danderson) [inline design/cncf-sandbox-proposal.md:61]: There is currently no trademark for MetalLB (not even an unregistered one, since we don't use (tm) anywhere. So, I don't know what assignment entails there, but I'm fine with whatever it does entail.

**Review by danderson — APPROVED** [review]
For the record: LGTM to whatever outcome, I defer to the current maintainers and will implement whatever they decide.

File: design/cncf-sandbox-proposal.md
Comment (russellb) [reply design/cncf-sandbox-proposal.md:61]: Thanks.  IANAL obviously ... I figured nothing had been registered but thought they may want you to sign something anyway, so figured I'd call it out.  I think you've clarified that it should be fine either way, so thanks!

Comment (danderson) [reply design/cncf-sandbox-proposal.md:61]: Yes, I'll happily assign whatever the CNCF lawyers deem appropriate, just wanted to clarify the current state of the world for the record, which is that no registered marks exist, and MetalLB has never used unregistered marks in communications. I have no idea what legal status that translates to, or what status CNCF wants, but they're welcome to all of it.

**Review by daxmc99 — APPROVED** [review]
LGTM, I think this would be an ideal home for the project.

**Review by rata — APPROVED** [review]
@russellb Thanks for the proposal, LGTM!

File: design/cncf-sandbox-proposal.md
Comment (johananl) [inline design/cncf-sandbox-proposal.md:29]:
```suggestion
- Transfer assets to a neutral third party (i.e. domains).
```

File: design/cncf-sandbox-proposal.md
Comment (johananl) [inline design/cncf-sandbox-proposal.md:35]:
```suggestion
- No changes to the MetalLB code are in scope of this proposal. This
```

File: design/cncf-sandbox-proposal.md
Comment (johananl) [inline design/cncf-sandbox-proposal.md:48]:
```suggestion
beneficial to MetalLB in the future. The CNCF web site does a nice job
```

File: design/cncf-sandbox-proposal.md
Comment (johananl) [inline design/cncf-sandbox-proposal.md:59]:
```suggestion
- (russellb) If approved, apply through the [proposal form](https://docs.google.com/forms/d/1bJhG1MuM981uQXcnBMv4Mj9yfV5_q5Kwk3qhBCLa_5A/viewform).
```

**Review by johananl — COMMENTED** [review]
Thanks for driving this @russellb!

I have one question: Do we know what the implications are in terms of new responsibilities, obligations, thins we aren't allowed to do etc. once the project moves to the CNCF? Since this seems like a one-way process, I want to ensure we are going to be happy with the new situation...

Added a few nits, too.

File: design/cncf-sandbox-proposal.md
Comment (russellb) [reply design/cncf-sandbox-proposal.md:60]: @danderson I think this is the form they'll ask you to sign: https://github.com/cncf/foundation/blob/master/agreements/CNCF%20Trademark%20and%20Account%20Assignment%20Agreement%20(2020%20-%20no%20reg%20trademarks).pdf

**Comment by russellb** [comment]
> I have one question: Do we know what the implications are in terms of new responsibilities, obligations, things we aren't allowed to do etc. once the project moves to the CNCF? Since this seems like a one-way process, I want to ensure we are going to be happy with the new situation...

Good question - it's definitely important that everyone understands and is comfortable with this!  It is indeed a one-way process.

The last time I went through this was for https://metal3.io.  Here was our issue for the onboarding process:  https://github.com/cncf/toc/issues/532

Not all of the onboarding steps required action on our part.  I beleive we're also already in compliance with the core requirements (open governance, compliance with the CNCF IP policy).  The initial work will roughly be:

* transfer of domains, logo.  There's also an [assignment](https://github.com/cncf/foundation/blob/master/agreements/CNCF%20Trademark%20and%20Account%20Assignment%20Agreement%20(2020%20-%20no%20reg%20trademarks).pdf) doc they will ask Dave to sign
* adding LF admins as github org owners (just in case...)
* updating our web site to add a CNCF footer
* getting set up with CNCF infrastructure (get accounts on their service portal, get a maintainers mailing list setup to receive info from the CNCF)

On an ongoing basis, the day-to-day impact is very minimal.  In my experience, I've seen:

* Invites to a monthly open forum with CNCF staff (not required to attend, but available if we have questions)
* Periodic emails, usually about upcoming events and invitations to participate as a project if we'd like to

**Comment by rata** [comment]
@johananl following the links in this doc (recursively) helped me to understand what is needed and provided by the CNCF, requirements for going from sandbox to incubation CNCF project, etc. I already closed the tabs, what @russellb mentions seems like a good summary, IMHO.

There is also some mention to the [Linux Foundation Antitrust policy][link] that we are expected to follow. Please read as much as you want/need to feel comfortable, it looked all sane to me if that helps in any way (and CNCF is the go-to for these projects, part of the linux foundation, it seems reasonable to be all sane :))

[link]: https://www.linuxfoundation.org/antitrust-policy

**Comment by johananl** [comment]
@rata @russellb I was referring mainly to "day 2" responsibilities, guidelines etc. I would like to ensure we aren't compromising our ability to effectively maintain the project, for example due to excess administrative overhead and/or being required to follow guidelines that somehow interfere with our role as maintainers.

I've done some reading:

- https://github.com/cncf/toc/blob/main/process/project_proposals.adoc
- https://github.com/cncf/toc/blob/main/process/graduation_criteria.adoc
- https://github.com/cncf/foundation/blob/master/code-of-conduct.md
- https://github.com/cncf/toc/blob/main/PRINCIPLES.md#no-kingmakers--one-size-does-not-fit-all
- https://github.com/cncf/foundation/blob/master/charter.md

All of the above seems reasonable to me. I haven't been able to find anything which makes me uneasy. I'm going to approve this PR and hope for the best :slightly_smiling_face: :crossed_fingers:

**Review by johananl — APPROVED** [review]

**Comment by russellb** [comment]
Thanks, everyone!  I'll merge this and proceed with the next step, which is to fill out a form and then wait for it to be reviewed.  I'll be sure to share updates as they come up.  Nothing is done-done until we hand over all the keys, so there's still plenty of time to probe some more if you think of anything.

**Pull request merged**
